grease/src/Grease/Setup.hs

```diff
@@ -95,31 +96,40 @@ safeSymbol = W4.safeSymbol . getValueName
 doLog :: MonadIO m => GreaseLogAction -> Diag.Diagnostic -> m ()
 doLog la diag = LJ.writeLog la (SetupDiagnostic diag)
 
-data SetupState sym ext argTys = SetupState
+
+-- | The result of setting up a pointer. `setupResPtr` represents the runtime value of the pointer and
+-- `setupResTgt` stores the runtime representation of what is stored in that pointer. These 
+-- results are memoized during setup based on `BlockId` so that a given block is only given 
+-- one runtime value.
+data SetupRes sym w = SetupRes { setupResPtr :: C.RegValue sym (Mem.LLVMPointerType w), setupResTgt ::  PtrTarget w (C.RegValue' sym)}
+
+data SetupState sym ext argTys w = SetupState
   { _setupMem :: Mem.MemImpl sym
   , _setupAnns :: Anns.Annotations sym ext argTys
+  -- Setup result for a given block setup
+  , _setupRes :: Map.Map BlockId (SetupRes sym w)
```

Comment:
I would suggest (1) making this a Haddock comment and (2) expanding it to reflect its intent, perhaps along these lines:
```suggestion
  -- | Memoization table for pointers based on the 'BlockId' in 'PtrTarget's.
  --
  -- See Haddock on 'PtrTarget' about 'BlockId's.
  , _setupRes :: Map.Map BlockId (SetupRes sym w)
```